clang/lib/Lex/PPCaching.cpp

```diff
@@ -116,20 +116,24 @@ const Token &Preprocessor::PeekAhead(unsigned N) {
 void Preprocessor::AnnotatePreviousCachedTokens(const Token &Tok) {
   assert(Tok.isAnnotation() && "Expected annotation token");
   assert(CachedLexPos != 0 && "Expected to have some cached tokens");
-  assert(CachedTokens[CachedLexPos-1].getLastLoc() == Tok.getAnnotationEndLoc()
-         && "The annotation should be until the most recent cached token");
+
+  // NOTE(pag): The split token tracker stuff might make this not true.
```

Comment:
Same here. May be we can protect the change with pasta flag and let assert be there if flag is not used.

Reply:
We don't really have a flag that can tell clang that pasta is doing its thing. This assert triggers because I change source locations inside of the split token tracker. Should I add a flag?

Reply:
If we don't have flag for spilt token callbacks then we can keep as it is.